jfr/src/main/java/io/netty/jfr/JfrChannelHandler.java

```diff
@@ -0,0 +1,417 @@
+    }
+
+    @Label("Close")
+    @Name("io.netty.channel.Close")
+    private static final class CloseEvent extends ChannelEvent {
+    }
+
+    @Label("Disconnect")
+    @Name("io.netty.channel.Disconnect")
+    private static final class DisconnectEvent extends ChannelEvent {
+    }
+
+    @Label("Connect")
+    @Name("io.netty.channel.Connect")
+    private static final class ConnectEvent extends ChannelEvent {
+        @Label("Remote Address")
+        String remoteAddress;
+        @Label("Local Address")
+        String localAddress;
+
+        void setRemoteAddress(SocketAddress remoteAddress) {
+            this.remoteAddress = String.valueOf(remoteAddress);
+        }
+
+        void setLocalAddress(SocketAddress localAddress) {
+            this.localAddress = String.valueOf(localAddress);
+        }
+    }
+
+    @Label("Channel Registered")
+    @Name("io.netty.channel.ChannelRegistered")
+    private static final class ChannelRegisteredEvent extends ChannelEvent {
+    }
+
+    @Label("Channel Unregistered")
+    @Name("io.netty.channel.ChannelUnregistered")
+    private static final class ChannelUnregisteredEvent extends ChannelEvent {
+    }
+
+    @Label("Channel Writability Changed")
+    @Name("io.netty.channel.ChannelWritabilityChanged")
+    private static final class ChannelWritabilityChangedEvent extends ChannelEvent {
+        @Label("Is Writable")
+        boolean isWritable;
+        @Label("Was Writable")
+        boolean wasWritable;
+
+        void setIsWritable(boolean writable) {
+            isWritable = writable;
+        }
+
+        void setWasWritable(boolean wasWritable) {
+            this.wasWritable = wasWritable;
+        }
+    }
+
+    @Label("Channel Inactive")
+    @Name("io.netty.channel.ChannelInactive")
+    private static final class ChannelInactiveEvent extends ChannelEvent {
+    }
+
+    @Label("Channel Active")
+    @Name("io.netty.channel.ChannelActive")
+    private static final class ChannelActiveEvent extends ChannelEvent {
+    }
+
+    @Label("Deregister")
+    @Name("io.netty.channel.Deregister")
+    private static final class DeregisterEvent extends ChannelEvent {
+    }
+
+    @Label("User Event Triggered")
+    @Name("io.netty.channel.UserEventTriggered")
+    private static final class UserEventTriggeredEvent extends ChannelEvent {
+        @Label("User Event")
+        String userEvent;
+
+        void setUserEvent(Object evt) {
+            this.userEvent = String.valueOf(evt);
+        }
+    }
+
+    @Label("Flush")
+    @Name("io.netty.channel.Flush")
+    private static final class FlushEvent extends ChannelIOEvent {
+    }
+
+    @Label("Exception Caught")
+    @Name("io.netty.channel.ExceptionCaught")
+    private static final class ExceptionCaughtEvent extends ChannelEvent {
+        @Label("Cause")
+        String cause;
+
+        void setCause(Throwable cause) {
+            this.cause = cause.toString();
```

Comment:
Invoking `toString` on the `Throwable` may have a noticeable performance impact.
I believe the best practice is to deconstruct the throwable with the message and the type.

```
@Label("Message")
String message;

@Label("Class")
Class<?> thrownClass;
```